lib/react_client/react_interop.dart

```diff
@@ -34,8 +36,54 @@ abstract class React {
 
   external static bool isValidElement(dynamic object);
   external static ReactClass get Fragment;
+
+  /// Returns a ref that can be attached to an element via the ref attribute.
+  ///
+  /// When a ref is passed to an element in render, a reference to the node becomes
+  /// accessible at the 'current' attribute of the ref.
+  ///
+  /// See: <https://reactjs.org/docs/refs-and-the-dom.html#creating-refs>.
+  external static createRef();
 }
 
+class createRef<CurrentType> {
+  var jsCreateRef;
+
+  createRef() {
+    jsCreateRef = React.createRef();
+  }
+
+  CurrentType get current {
+    var jsCurrent = getProperty(jsCreateRef, 'current');
+    if (jsCurrent != null) {
+      if (hasProperty(jsCurrent, 'dartComponent')) {
+        return getProperty(jsCurrent, 'dartComponent');
+      }
+    }
+    return jsCurrent;
```

Comment:
This may be a #superNit, but would it be worthwhile to move `return jsCurrent` into the if block starting on line 58? As is, there is a chance of returning `null` under the guise of `jsCurrent`, which seems like it could be a little insidious. 

If `null` is an acceptable return value (not worthy of throwing), then this: 
```
    if (jsCurrent != null) {
      if (hasProperty(jsCurrent, 'dartComponent')) {
        return getProperty(jsCurrent, 'dartComponent');
      }

      return jsCurrent;
    }
    return null;
```
seems more declarative to what's could actually be happening. Definitely just a suggestion though!

Reply:
That makes sense. I ended up removing the null check in the if statement and just checking if `jsCurrent` is null while getting the `dartComponent` property. I can still add that null return if you still think it’s necessary, but it will add another layer of if statements or an if statement at the top to return null if `jsCurrent` is null. Let me know what you think!

Reply:
Nope, it was convenient before but with the additional specificity I think it's great as is!